src/mat73.c

```diff
@@ -651,6 +651,8 @@ Mat_H5ReadFieldNames(matvar_t *matvar, hid_t dset_id, hsize_t *nfields)
     } else {
         err = MATIO_E_NO_ERROR;
     }
+    if (*nfields == 0)
+        *nfields = 1;
```

Comment:
```suggestion
    if ( *nfields == NULL ) {
        *nfields = 1;
    }
```

Thanks for the fix. Please do not forget to run clang-format.